---
title: "story_time"
author: "Jordan Whitlock"
date: '2023-06-13'
output: html_document
---

Goal: Investigate Gfap+ astrocytes within the context of neurodegeneration in order to determine if we are seeing a similar signature of reactive astrocytes found in Alzheimer's disease, Parkinsons, or other neurodegenerative disorders within the context of SGS.

Hypothesis: Gfap+ (reactive astrocytes) are specific/significantly differentially expressed in SGS astrocytes compared to control and could therefore be contributing in a similar manner to neurodegeneration in SGS as found previously with AD. 

Steps:
1. Subset the astrocytes specifically
2. Recluster them to ID subclusters
3. Use FindAllMarkers to ID statistically significant DEGs between subclusters. 
* Gfap is returned as stat significant marker of sub cluster 7 compared to all other astrocyte subclusters.
4. Use FindMarkers to ID statistically significant DEGs between conditions.
* Gfap is not returned as between the two conditions. 

Conclusion: While Gfap is not a DEG between conditions (S858R for SGS and WT controls) it is present as a subcluster in my astrocytes as a whole in both conditions. In conclusion, while this does not definitively point out a major difference in Gfap+ astrocytes between conditions as hoped, this does not rule out astrocytes as a whole in contributing to the neurodegeneration present in kiddos with SGS. 

```{r}
set.seed(2178)
library(Seurat)
library(here)
library(clustree)
ptm <- proc.time()
```


```{r}
load(here("data/setbp1_cerebralintcelltypes.Rdata"))

# looking at Gfap
FeaturePlot(cerebral_int_celltypes, features = "Gfap",
            reduction = "umap_harmony",
            label = TRUE, split.by = "type")

# extracting astrocytes for both conditions
subset <- subset(cerebral_int_celltypes, idents = "astrocytes")

p1 <- DimPlot(subset,
  reduction = "umap_harmony",
  group.by = "seurat_clusters",
  label = TRUE
)
p2 <- DimPlot(subset,
  reduction = "umap_harmony",
  group.by = "orig.ident",
  label = TRUE
)
p3 <- DimPlot(subset,
  reduction = "umap_harmony",
  group.by = "type",
  label = TRUE
)
p1
p2
p3

subset <- FindNeighbors(subset, dims = 1:30, reduction = "harmony")
subset <- FindClusters(subset,
  graph.name = "RNA_snn",
  resolution = 0.9,
  algorithm = 4,
  method = "igraph"
)

clustree(subset@meta.data, prefix = "RNA_snn_res.")

cerebral_markers_genes <- FindAllMarkers(subset,
  log2FC.threshold = 0.2, test.use = "wilcox",
  min.pct = 0.1, min.diff.pct = 0.2, only.pos = TRUE, max.cells.per.ident = 50,
  assay = "RNA"
)

# visualize markers at 0.9 resolution
p4 <- DimPlot(subset,
  reduction = "umap_harmony",
  group.by = "type",
  label = TRUE
)

png(
  file = here("results/module_scores/reactive_astrocytes_subcluster.png"),
  width = 4000,
  height = 3000,
  res = 250
)
FeaturePlot(subset, features = "Gfap", reduction = "umap_harmony", label = TRUE,
            split.by = "type")
dev.off()

# seeing if Gfap is DE in cluster 7 between conditions
reactive_subset <- subset(subset, idents = 7)
Idents(reactive_subset) <- reactive_subset$type

reactive_markers <- FindMarkers(reactive_subset, ident.1 = "heterozygous",
                                ident.2 = "control")
reactive_markers$gene <- rownames(reactive_markers)
```

Important note from [Patani et al Nature Reviews Neurology 2023](https://www.nature.com/articles/s41582-023-00822-1): 

"Many of the published scRNA-seq and snRNA-seq data sets capture large numbers of individual cells but have been underpowered for astrocytes, often with only a few dozen of these cells being collected from each individual animal. Combined with the low sequence depth afforded by these methods, this has led to a number of attempts at gene clustering that have been unable to define differentially expressed genes that are representative of specific astrocyte populations in vivo. This artefact of clustering driven by low numbers of astrocytes has also caused several reactive astrocyte sub-states to be missed owing to low abundance50. This is an important limitation as even some well-accepted reactive astrocyte states, such as scar-forming astrocytes, represent less than 5% of all astrocytes in the CNS. Interpretation of weakly powered scRNA-seq and snRNA-seq data sets has often led to the erroneous conclusion that transcriptional changes associated with astrocyte reactivity are purely disease specific72, in contrast to the homogeneity of some sub-states of microglia73 and oligodendrocytes56,74 across diseases. These latter highly powered studies56,73,74 show that, although both microglia and oligodendrocytes have considerable transcriptomic heterogeneity within a disease or brain region, some reactive sub-states are common across disease states. Several of these microglial and oligodendrocyte states are also reported following inflammation (a common response to nearly all neurodegenerative diseases, infections and traumas). In addition, the gene expression profiles that are observed in these sub-states have some overlap with those of proliferative scar-forming reactive astrocytes or astrocytes that are associated with AÂ² plaques75,76. Further studies focusing on the integration of multiple data sets from different disease models and/or patient cohorts are likely to further validate the presence of sub-states of reactive astrocytes across different diseases as each additional data set included in such meta-analyses will further increase astrocyte numbers and, hence, the power of downstream interpretation"


```{r}
fptm <- proc.time() - ptm
fptm <- (fptm[3] / 60) / 60
print(paste0("Run time: ", fptm, " hours"))
```
[1] "Run time: 0.149947777777778 hours"

```{r}
style_file(here("src/module_scores/03_Setbp1_reactive_astrocytes.Rmd")) # Note: lintr was run as well
```

```{r}
sessionInfo()
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] clustree_0.5.0     ggraph_2.1.0       ggplot2_3.4.1      here_1.0.1         SeuratObject_4.1.3 Seurat_4.3.0      

loaded via a namespace (and not attached):
  [1] Rtsne_0.16             colorspace_2.1-0       deldir_1.0-6           ellipsis_0.3.2         ggridges_0.5.4         rprojroot_2.0.3       
  [7] rstudioapi_0.14        spatstat.data_3.0-0    farver_2.1.1           leiden_0.4.3           listenv_0.9.0          graphlayouts_0.8.4    
 [13] ggrepel_0.9.3          fansi_1.0.4            codetools_0.2-18       splines_4.1.3          knitr_1.42             polyclip_1.10-4       
 [19] jsonlite_1.8.4         ica_1.0-3              cluster_2.1.2          png_0.1-8              uwot_0.1.14            ggforce_0.4.1         
 [25] shiny_1.7.4            sctransform_0.3.5      spatstat.sparse_3.0-0  compiler_4.1.3         httr_1.4.5             backports_1.4.1       
 [31] Matrix_1.5-3           fastmap_1.1.1          lazyeval_0.2.2         limma_3.50.3           cli_3.6.1              later_1.3.1           
 [37] tweenr_2.0.2           htmltools_0.5.4        tools_4.1.3            igraph_1.4.3           gtable_0.3.1           glue_1.6.2            
 [43] RANN_2.6.1             reshape2_1.4.4         dplyr_1.1.2            rappdirs_0.3.3         Rcpp_1.0.10            scattermore_0.8       
 [49] vctrs_0.6.2            spatstat.explore_3.0-6 nlme_3.1-155           progressr_0.13.0       lmtest_0.9-40          spatstat.random_3.1-3 
 [55] xfun_0.37              stringr_1.5.0          globals_0.16.2         mime_0.12              miniUI_0.1.1.1         lifecycle_1.0.3       
 [61] irlba_2.3.5.1          goftest_1.2-3          future_1.31.0          MASS_7.3-55            zoo_1.8-11             scales_1.2.1          
 [67] tidygraph_1.2.3        promises_1.2.0.1       spatstat.utils_3.0-1   parallel_4.1.3         RColorBrewer_1.1-3     yaml_2.3.7            
 [73] reticulate_1.28        pbapply_1.7-0          gridExtra_2.3          stringi_1.7.12         checkmate_2.1.0        rlang_1.1.1           
 [79] pkgconfig_2.0.3        matrixStats_0.63.0     evaluate_0.20          lattice_0.20-45        ROCR_1.0-11            purrr_1.0.1           
 [85] tensor_1.5             labeling_0.4.2         patchwork_1.1.2        htmlwidgets_1.6.1      cowplot_1.1.1          tidyselect_1.2.0      
 [91] parallelly_1.34.0      RcppAnnoy_0.0.20       plyr_1.8.8             magrittr_2.0.3         R6_2.5.1               generics_0.1.3        
 [97] pillar_1.9.0           withr_2.5.0            fitdistrplus_1.1-8     survival_3.3-1         abind_1.4-5            sp_1.6-0              
[103] tibble_3.2.1           future.apply_1.10.0    KernSmooth_2.23-20     utf8_1.2.3             spatstat.geom_3.0-6    plotly_4.10.1         
[109] rmarkdown_2.20         viridis_0.6.2          grid_4.1.3             data.table_1.14.8      digest_0.6.31          xtable_1.8-4          
[115] tidyr_1.3.0            httpuv_1.6.11          munsell_0.5.0          viridisLite_0.4.1     