---
title: "Setbp1_Cortex_subclustering"
output: html_document
date: '2023-02-23'
---

While performing gene targeting score analysis looking across cell types (https://docs.google.com/document/d/17ZsaiF8XBK99MS7dLIxvIvWpC59OAmSlRszeSVskUgY/edit?usp=sharing), it was noted there is a bimodal presence of targeting scores in oligodendrocytes could suggest that there is a subcelltype in the oligodendrocytes in the disease condition compared to control. Paper provided by Tabea, where she explained this is something that has been seen in AD with oligos (Single-cell RNA-sequencing identifies disease-associated oligodendrocytes in male APP NL-G-F and 5XFAD mice). With respect to gene targeting across cell type, I see an opposite shift where mural cells go from bimodal in controls to unimodal in S858R mice, suggesting there is some loss of nuclei or change in the genetic makeup of the nuclei population. In order to investigate this, I am looking at the potential presence of subclusters within the final, annotated cerebral cortex seurat object below.

```{r load-libraries}
set.seed(2178)
library(Seurat)
library(clustree)
library(here)
library(styler)
library(lintr)
```

```{r load-object}
load(here("data/setbp1_cerebralintcelltypes.Rdata"))
select_resolution <- "RNA_snn_res.0.9"
cerebral_int_celltypes <- SetIdent(cerebral_int_celltypes, 
                                   value = select_resolution)

DimPlot(cerebral_int_celltypes, 
        reduction = "umap_harmony", 
        group.by = "cell_type")

Idents(cerebral_int_celltypes) <- "cell_type"
```

Note: When prompted about miniconda, reply "N".
```{r subcluster-oligo}
# Subset data.int.celltypes cluster1 and sub-cluster
subset <- subset(cerebral_int_celltypes, idents = "oligodendrocytes")

p1 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "seurat_clusters", 
              label = TRUE)
p2 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "orig.ident", 
              label = TRUE)
p3 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "type", 
              label = TRUE)

subset <- FindNeighbors(subset, dims = 1:30, reduction = "harmony")
subset <- FindClusters(subset, 
                       graph.name = "RNA_snn", 
                       resolution = 0.9, 
                       algorithm = 4, 
                       method = "igraph")

clustree(subset@meta.data, prefix = "RNA_snn_res.")

cerebral_markers_genes <- FindAllMarkers(subset,
  log2FC.threshold = 0.2, test.use = "wilcox",
  min.pct = 0.1, min.diff.pct = 0.2, only.pos = TRUE, max.cells.per.ident = 50,
  assay = "RNA"
)

# visualize markers at 0.9 resolution
p4 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "seurat_clusters", 
              label = TRUE)

png(here("results/seurat/Cortex_subclustering/Setbp1_cortex_oligo_subcluster_UMAP.png"), 
    width = 1000, height = 500)
(p1 + p2) / (p3 + p4)
dev.off
```


```{r subcluster-mural}
# Subset data.int.celltypes cluster1 and sub-cluster
subset <- subset(cerebral_int_celltypes, idents = "mural.cells")

png(here("results/seurat/Cortex_subclustering/Setbp1_cortex_mural_subcluster_seuratclusterUMAP.png"), 
    width = 1000, height = 500)
p1 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "seurat_clusters", 
              label = TRUE)
p1
dev.off()

png(here("results/seurat/Cortex_subclustering/Setbp1_cortex_mural_subcluster_identUMAP.png"), 
    width = 1000, height = 500)
p2 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "orig.ident", 
              label = FALSE)
p2
dev.off()

png(here("results/seurat/Cortex_subclustering/Setbp1_cortex_mural_subcluster_conditionUMAP.png"), 
    width = 1000, height = 500)
p3 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "type", 
              label = FALSE)
p3
dev.off()

subset <- FindNeighbors(subset, dims = 1:30, reduction = "harmony")
subset <- FindClusters(subset, 
                       graph.name = "RNA_snn", 
                       resolution = 0.9, 
                       algorithm = 4, 
                       method = "igraph")

clustree(subset@meta.data, prefix = "RNA_snn_res.")

cerebral_markers_genes <- FindAllMarkers(subset,
  log2FC.threshold = 0.2, test.use = "wilcox",
  min.pct = 0.1, min.diff.pct = 0.2, only.pos = TRUE, max.cells.per.ident = 50,
  assay = "RNA"
)

# visualize markers at 0.9 resolution
png(here("results/seurat/Cortex_subclustering/Setbp1_cortex_mural_subcluster_UMAP.png"), 
    width = 1000, height = 500)
p4 <- DimPlot(subset, 
              reduction = "umap_harmony", 
              group.by = "seurat_clusters", 
              label = FALSE)
p4
dev.off()
```

The above shows no distinct grouping between conditions or samples when looking at subclustering. It is likely that the difference in targeting scores is either driven by biology with respect to regulation, or an artifact of differences in sequencing or read depth. 

```{r}
sessionInfo()
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lintr_3.0.2        styler_1.9.0       here_1.0.1         clustree_0.5.0     ggraph_2.1.0       ggplot2_3.4.1      SeuratObject_4.1.3 Seurat_4.3.0      

loaded via a namespace (and not attached):
  [1] Rtsne_0.16             colorspace_2.1-0       deldir_1.0-6           ellipsis_0.3.2         ggridges_0.5.4         rprojroot_2.0.3        rstudioapi_0.14       
  [8] spatstat.data_3.0-0    leiden_0.4.3           listenv_0.9.0          farver_2.1.1           remotes_2.4.2          graphlayouts_0.8.4     ggrepel_0.9.3         
 [15] fansi_1.0.4            xml2_1.3.3             R.methodsS3_1.8.2      codetools_0.2-18       splines_4.1.3          knitr_1.42             polyclip_1.10-4       
 [22] jsonlite_1.8.4         ica_1.0-3              cluster_2.1.2          R.oo_1.25.0            png_0.1-8              uwot_0.1.14            ggforce_0.4.1         
 [29] shiny_1.7.4            sctransform_0.3.5      spatstat.sparse_3.0-0  compiler_4.1.3         httr_1.4.5             backports_1.4.1        Matrix_1.5-3          
 [36] fastmap_1.1.1          lazyeval_0.2.2         limma_3.50.3           cli_3.6.0              later_1.3.0            tweenr_2.0.2           htmltools_0.5.4       
 [43] tools_4.1.3            igraph_1.4.1           gtable_0.3.1           glue_1.6.2             RANN_2.6.1             reshape2_1.4.4         dplyr_1.1.0           
 [50] rappdirs_0.3.3         Rcpp_1.0.10            scattermore_0.8        vctrs_0.5.2            spatstat.explore_3.0-6 nlme_3.1-155           progressr_0.13.0      
 [57] lmtest_0.9-40          spatstat.random_3.1-3  xfun_0.37              stringr_1.5.0          ps_1.7.2               globals_0.16.2         mime_0.12             
 [64] miniUI_0.1.1.1         lifecycle_1.0.3        irlba_2.3.5.1          goftest_1.2-3          future_1.31.0          MASS_7.3-55            zoo_1.8-11            
 [71] scales_1.2.1           tidygraph_1.2.3        promises_1.2.0.1       spatstat.utils_3.0-1   rex_1.2.1              parallel_4.1.3         RColorBrewer_1.1-3    
 [78] reticulate_1.28        pbapply_1.7-0          gridExtra_2.3          stringi_1.7.12         desc_1.4.2             checkmate_2.1.0        cyclocomp_1.1.0       
 [85] rlang_1.0.6            pkgconfig_2.0.3        matrixStats_0.63.0     lattice_0.20-45        ROCR_1.0-11            purrr_1.0.1            tensor_1.5            
 [92] labeling_0.4.2         patchwork_1.1.2        htmlwidgets_1.6.1      processx_3.8.0         cowplot_1.1.1          tidyselect_1.2.0       parallelly_1.34.0     
 [99] RcppAnnoy_0.0.20       plyr_1.8.8             magrittr_2.0.3         R6_2.5.1               generics_0.1.3         pillar_1.8.1           withr_2.5.0           
[106] fitdistrplus_1.1-8     survival_3.3-1         abind_1.4-5            sp_1.6-0               tibble_3.1.8           future.apply_1.10.0    crayon_1.5.2          
[113] KernSmooth_2.23-20     utf8_1.2.3             spatstat.geom_3.0-6    plotly_4.10.1          viridis_0.6.2          grid_4.1.3             data.table_1.14.8     
[120] callr_3.7.3            digest_0.6.31          R.cache_0.16.0         xtable_1.8-4           tidyr_1.3.0            httpuv_1.6.9           R.utils_2.12.2        
[127] munsell_0.5.0          viridisLite_0.4.1     

```{r}
# run style
style_file(here("src/seurat_scripts/Setbp1_Cortex_subclustering.Rmd"))
# lintr was run as well
```



